phpstan.neon.dist

```diff
@@ -63,6 +63,15 @@ parameters:
               - %currentWorkingDirectory%/core-bundle/src/Controller/AbstractController.php
               - %currentWorkingDirectory%/core-bundle/src/Command/MigrateCommand.php
 
+        # Ignore that services via a locator are falsely identified as private/not registered.
+        - message: '#Service ".*" is not registered in the container\.#'
+          paths:
+              - %currentWorkingDirectory%/core-bundle/src/Image/Studio/*.php
+
+        - message: '#Service "parameter_bag" is private\.#'
```

Comment:
Not sure if suppressing the error is better than fixing it (if possible)?

Reply:
I don't think we can fix it. It's a lack of PHPStan understanding service locators.

see https://github.com/phpstan/phpstan-symfony/issues/76

Reply:
See https://github.com/contao/contao/pull/1753#discussion_r455644320